src/include/txn_inline.h

```diff
@@ -1236,7 +1236,8 @@ __wt_txn_update_check(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE
     txn = session->txn;
     txn_global = &S2C(session)->txn_global;
 
-    if (txn->isolation != WT_ISO_SNAPSHOT)
+    /* Don't check if transaction isolation is not snapshot or the table is metadata. */
+    if (txn->isolation != WT_ISO_SNAPSHOT || WT_IS_METADATA(cbt->dhandle))
```

Comment:
I'm not clear on why we need this change. We set the metadata transaction to read-committed below so it should succeed on the `txn->isolation != WT_ISO_SNAPSHOT` conditional already.

Reply:
The code flow doesn't come from the meta ckpt session, it is coming from bulk cursor close checkpoint. Added more details in Jira.